Zend/zend_API.h

```diff
@@ -636,6 +636,20 @@ END_EXTERN_C()
 		ZVAL_PSTRINGL(z, "", 0);				\
 	} while (0)
 
+#define ZVAL_ONE_CHAR_STRING(z, c)  do {		\
```

Comment:
what about using `ZVAL_ONE_CHAR_INTERNED_STR`? So that we wouldn't lose the information that it creates an interned string?

Reply:
> what about using `ZVAL_ONE_CHAR_INTERNED_STR`? So that we wouldn't lose the information that it creates an interned string?

but API named `*_STR` usually takes `zend_string *` as a parameter and `ZVAL_ONE_CHAR_INTERNED_STRING` is too long...

Reply:
Length of a macro name is not a valid argument IMHO, readability is more important than just how many characters less you will type once.

Reply:
> Length of a macro name is not a valid argument IMHO, readability is more important than just how many characters less you will type once.

You're right, I prefer `ZVAL_ONE_CHAR_INTERNED_STRING`

Reply:
@twose Using `str` vs `string` has a semantic difference. Please see the difference here: https://github.com/php/php-src/pull/5647#discussion_r432889879 Initially, I also argued against using abbreviations, but "unfortunately" they make sense in this case. :)

Reply:
@kocsismate I'm a little confused now, this macro accepts a parameter of type `char`, not `zend_string *`, so we named it `*_ONE_CHAR_INTERNED_STRING` instead of `ONE_CHAR_INTERNED_STR`, is that right?

Reply:
I'd suggest using just `ZVAL_CHAR(zv, c)` ... :)

Reply:
> I'd suggest using just `ZVAL_CHAR(zv, c)` ... :)

I'm in favor of your proposal, the reason I didn’t do this at first was that there is no `char` type in PHP